test/spec/directives/bsSwitchSpec.js

```diff
@@ -723,4 +723,140 @@ describe('Directive: bsSwitch', function () {
   }
   it('should evaluate change expression when model changes', inject(makeTestModelSwitchChange()));
   it('should evaluate change expression when model changes', inject(makeTestModelSwitchChange(true)));
+
+  // Test the null model from true state
+  function makeTestToIndeterminateNullFromTrue(input) {
+    return function () {
+      var element = compileDirective(undefined, input);
+      expect(element.hasClass(CONST.SWITCH_OFF_CLASS)).toBeFalsy();
+      expect(element.hasClass(CONST.SWITCH_ON_CLASS)).toBeTruthy();
+      scope.model = true;
+      scope.$apply();
+      expect(element.hasClass(CONST.SWITCH_INDETERMINATE_CLASS)).toBeFalsy();
+      expect(element.hasClass(CONST.SWITCH_ON_CLASS)).toBeTruthy();
+      expect(element.hasClass(CONST.SWITCH_OFF_CLASS)).toBeFalsy();
+      scope.model = null;
+      scope.$apply();
+      expect(element.hasClass(CONST.SWITCH_INDETERMINATE_CLASS)).toBeTruthy();
+      expect(element.hasClass(CONST.SWITCH_OFF_CLASS)).toBeFalsy();
+      expect(element.hasClass(CONST.SWITCH_ON_CLASS)).toBeTruthy();
```

Comment:
I thought that's how your library behaves?
All your test codes were written that way.